packages/@react-spectrum/utils/src/styleProps.ts

```diff
@@ -166,7 +166,9 @@ function borderColorValue(value: BorderColorValue, version = 5) {
 }
 
 function borderSizeValue(value: BorderSizeValue) {
-  return `var(--spectrum-alias-border-size-${value})`;
+  return value && value !== 'none'
+    ? `var(--spectrum-alias-border-size-${value})`
+    : undefined;
```

Comment:
this won't work, border size defaults to `medium`
https://jsfiddle.net/qv5kp0zr/1/

We'll need to set the returned value to 0 instead

Reply:
Good catch. While testing it was working for me on storybook but it is better to return `0`. I've updated the PR. Thanks